boot/makebootable.go

```diff
@@ -280,21 +280,42 @@ func makeBootable20RunMode(model *asserts.Model, rootdir string, bootWith *Boota
 	blVars := map[string]string{
 		"kernel_status": "",
 	}
-	if err := bl.SetBootVars(blVars); err != nil {
-		return fmt.Errorf("cannot set run system environment: %v", err)
-	}
 
 	ebl, ok := bl.(bootloader.ExtractedRunKernelImageBootloader)
-	if !ok {
-		return fmt.Errorf("cannot use %s bootloader: does not support extracted run kernel images", bl.Name())
-	}
+	if ok {
+		// we can use ebl, so just set the EnableKernel after setting
+		// kernel_status
+		if err := bl.SetBootVars(blVars); err != nil {
+			return fmt.Errorf("cannot set run system environment: %v", err)
+		}
 
-	if err := ebl.EnableKernel(bootWith.Kernel); err != nil {
-		return err
+		if err := ebl.EnableKernel(bootWith.Kernel); err != nil {
+			return err
+		}
+	} else {
+		// TODO:UC20: should we make this more explicit with a new
+		//            bootloader interface that is checked for first before
+		//            ExtractedRunKernelImageBootloader the same way we do with
+		//            ExtractedRecoveryKernelImageBootloader?
```

Comment:
Maybe something in the lines of `Extracted*ImplicitKernelImage` since we do not mention the kernel name? Whereas, for bootloaders not supporting this interface, we must explicitly name the kernel in the bootloader variables.

Reply:
I think we should leave this as TODO for now, historically I guess we don't want individual bootloaders to know about their bootenv variables, so this becomes a bit tricky.

Reply:
I think this might get a bit clearer once we try to support lk